src/testdir/test_visual.vim

```diff
@@ -1099,6 +1099,49 @@ func Test_exclusive_selection()
   bw!
 endfunc
 
+" Test for inclusive motion in visual mode with 'exclusive' selection
+func Test_inclusive_motion_selection_exclusive()
+  func s:compare_exclu_inclu(line, keys, expected_exclu)
+    call setline(1, a:line)
+    set selection=exclusive
+    call feedkeys("\<Esc>" . a:keys, 'xt')
```

Comment:
There is no need to enter `\<Esc>` in normal mode. There is no need to enter "just in case". It may erase the problem.
The same applies to all subsequent `\<Esc>`.